Week1 [STEP 2] yyjjgg

yyjjgg opened the pull request:
Week1 [STEP2]

  - 이번 STEP2를 진행할때는 코드 작성 보다는 순서도 작성에서 많은 시간을 할애했습니다.
     그래서 순서도 관련 학습내용과 위키백과를 참고하여 아래의 순서도와 같이 작성을 하였습니다.

@hrjy6278

Commit:
week1_step2_첫번째 커밋

hrjy6278 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,19 @@
 import Foundation
 
-print("Hello, World!")
+func icecream(){
```
현재 함수명 `incream` 만으로는 해당 함수가 어떤 `행동`을 하는지 알 수 가 없는 것 같아요.
함수는 어떠한 특정 `행동`을 하기 위한 존재입니다. 
따라서 함수,메서드는 보통 명사로 시작하지 않고 동사로 시작합니다.
https://www.swift.org/documentation/api-design-guidelines/ 
위 링크를 참조하셔서 네이밍 가이드를 한번 읽어보시고 함수 이름을 더 명확하게 해주셨으면 좋겠습니다. 😉

yyjjgg replied:
네이밍 가이드를 읽은 후 함수명을 변경하였습니다.
icecream은 drawIcecream으로, stick은 drawStick으로 변경하였습니다.

hrjy6278 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,19 @@
 
 import Foundation
 
-print("Hello, World!")
```
코드 삭제 처리 좋습니다 💯

yyjjgg replied:
감사합니다!

hrjy6278 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,19 @@
 
-print("Hello, World!")
+func icecream(){
+    for count1 in 1...8 {
```
반복문을 사용하셨네요 👏🏻👏🏻👏🏻👏🏻
하지만 여기서 `count1`은 for 문 안에서 전혀 사용되지 않고 있는 것 같아요.
아마 Xcode에서 **주황색 문구가** 뜨지 않으셨나요? 
문구가 뜨셨으면 그대로 해결해 보시고 해결한 결과가 어떤건지 저에게 알려주세요!

yyjjgg replied:
처음엔 주황색 문구를 봐도 무슨 말인지 이해를 못하였어요.
그래서 반복문 관련 학습자료를 다시 한번 복습하고 count를 drawIcecream 함수와 drawStick 함수의 매개변수로 설정하고서
for문을 함수 내부에서 제거 후 함수 바깥에서 생성하였더니 주황색 문구도 사라지고 아이스크림 모양이
이전과 같은 모양으로 작성되었습니다!

hrjy6278 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,19 @@
+}
 
+func stick(){
+    for count2 in 1...4 {
```
여기도 `incream` 함수의 반복문과 동일합니다!

yyjjgg replied:
네!
혹시 count1과 count2를 구분짓지 않고 count 한가지로 사용해서
drawIcecream과 drawStick에 각각 같은 이름의 count를 매개변수로 선언해도 되는것인가요?
코드를 실행할때는 문제가 없는데 이름이 같으면 문제가 있는지 궁금합니다.

hrjy6278 replied:
네 해당 count는 반복문의 현재 Loop의 숫자를 알려주게 됩니다.  count 반복문 안에서만 사용할 수 있기 때문에 말씀해주신대로 반복문에 각각 count를 할당 해 주셔도 문제가 없습니다!

이렇게 생각해보시면 좋을 것 같아요.
```
for count in 1...5 { 
count는 이 안에서만 사용가능합니다.
이 부분을 벗어나게 되면 어디서도 사용이 불가합니다.
} 
```

다만 현재 count를 정해두시고 함수 내부에서 전혀 쓰이고 있질않죠? 이럴때 wildcard pattern 을 좀 더 공부해보시길 추천드리겠습니다.
https://docs.swift.org/swift-book/ReferenceManual/Patterns.html 
이 링크로 가셔서 Wildcard Pattern을 보시면 됩니다.🙂

yyjjgg replied:
wildcard pattern을 이용하여 사용되지 않던 count를 제거하였습니다!

hrjy6278 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,19 @@
 import Foundation
 
-print("Hello, World!")
+func icecream(){
```
함수 내부 구문의 시작 braces인 `{ }` 와 반복문의 시작  `{ }` 의 공간이 서로 다른 것 같아요. 
이부분은 준근님이 맞게 하나로 통일해 주셨으면 좋겠습니다.

저는 개인적으로 `func someFunction() { }`  처럼 `() {}` 사이에 한칸을 띄우는 편입니다.

yyjjgg replied:
서로 다른 부분을 확인하고 () {} 사이에 한칸을 띄우는 방식으로 통일하여 변경하였습니다!

hrjy6278 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,19 @@
+    }
+}
+
+
```
현재 함수의 종료 시점과 함수의 실행부분이 Line 두칸이 띄워진게 보입니다. 

이것은 의도하신 걸 까요?

yyjjgg replied:
의도했던 것이 아니여서 Line을 한칸만 띄우도록 변경하였습니다

hrjy6278 reviewed (changes requested):
@yyjjgg  준근님 안녕하세요! 스텝2를 정말 빠르게 PR을 보내주셨네요. 👏🏻 👏🏻 👏🏻 👏🏻 
스텝2의 요구사항 대로 아이스크림을 정확히 출력한 점 정말 대단하십니다. 💯💯
하지만 몇가지 아쉬운 부분도 보였는데요. 이부분은 제가 코멘트를 남겨 놓았으니 코멘트를 확인해 주시면 될 것 같습니다!

그리고 현재 Commit된 게 하나인데요. 이부분을 조금 더 나눠서 commit을 해주셨으면 좋겠습니다.
저는 개인적으로 함수 하나당(하나의 기능) 한개의 커밋을 선호하는 편입니다.
또한 커밋 메시지만으로는 현재 어떤 기능이 바뀌고, 수정되고 추가됐는지 전혀 알 수가 없습니다. 만약 이 커밋이 실전과 같은 커밋이라면
동료들은 커밋의 내용을 전부 다 보고 나서야 이 커밋이 어떤 기능이추가되고, 업데이트가 된건지 알 수 있겠죠?
구글에 `git commit convention` 을 검색하시면 많은 컨벤션이 나오니 해당 컨벤션을 보시고 맘에드는 컨벤션으로 git commit Message를 맞춰주시는것도 좋아보입니다!

마지막으로 순서도를 보자면 함수의 실행부분이 전혀 안보이는 것 같아요...😗
순서도를 보자면 하나의 함수가 순서도의 역활을 다 담당하고 있는 것 같아요.
하지만 코드에서는 함수가 두개로 나눠져있죠?
이것 처럼 순서도도 마찬가지로 함수실행 하는 부분이 추가 되었으면 좋을 것 같아요!!

Commit:
refactor : 불필요한 Line 삭제

Commit:
refactor : 함수와 반복문의 시작{ }의 공간 통일 및 재배치

Commit:
refactor : 함수명 변경

icecream -> drawIcecream , stick -> drawStick

Commit:
Remove : 코드 제거

icecream() 코드 제거
stick() 코드 제거

Commit:
Update : 함수 내용 변경 및 반복문 작성

drawIcecream 함수와 drawStick 함수 안에 있던 for-in 반복문을 제거하고
각 함수에 count라는 매개변수를 생성하였다.
그리고 2개의 for-in 반복문을 생성하여
각각 drawIcecream 함수 8회 반복, drawStick 함수 4회 반복 하도록
코드를 변경하였다.

Commit:
Add : 순서도 추가

Week1 [STEP 2] 아이스크림 만들기 순서도 재작성 후 추가

hrjy6278 reviewed (approved):
수정이 많이 되었네요 바쁘실텐데 시간내셔서 공부하시는 모습이 멋집니다👍
커밋 컨벤션도 보기 좋게 바꾸셨고, 정말 칭찬드리고 싶은거는 커밋을 기능별로 나누신점 👏🏻👏🏻👏🏻👏🏻👏🏻👏🏻👏🏻 박수 백번입니다. 
그리고 이런 습관은 나중에 Commit을 되돌릴 때 아주 유용할 거예요!

질문주신 것에 대해 코멘트를 남겨드릴게요!

> 코멘트해주신 부분에 대해서 코드 변경 후 커밋 등을 진행했는데요
기존에 PR이 된 상태에서는 제가 커밋 이후에 푸쉬까지 하면 별도의 수정 없이
자동으로 PR에 내용이 변경되는 것이 맞나요?

- 넵 Commit Push 까지하시면 변경사항이 자동으로 PR에 반영됩니다. 아주 잘하셨어요. 👍👍👍
 지금처럼만 앞으로 계속 해주시면 됩니다!!!

> 순서도에 대해 전반적으로 이해하는데 어려움이 있었습니다.
새롭게 작성한 순서도 역시 고칠 점이 있다면 다시 한번 말씀해주세요!
- 순서도를 보자면 현재 count가 8보다 작은가 ? -> No 시에 `count = 1` 부분은 없어도 될 것 같아보여요. 아니면 함수 drawStick 호출 다음에 있어야 하지 않을까 라는 생각이 듭니다. 하지만 이 부분은 크게 중요한 부분은 아닙니다!
